index.bs

```diff
@@ -368,6 +370,160 @@ dictionary SanitizerConfig {
+        there may be no duplicate entries between these lists.
+    * If a local {{SanitizerElementNamespaceWithAttributes/removeAttributes}} remove lists exists,
+        then all its entries must also be listed in the global {{SanitizerConfig/attributes}}
+        allow list.
+    * If {{SanitizerConfig/dataAttributes}} is true, then no [=custom data attributes=] may be
+        listed in any of the allow-lists.
+  * If a global {{SanitizerConfig/removeAttributes}} remove list exists, then:
+    * If a local {{SanitizerElementNamespaceWithAttributes/attributes}} allow lists exists,
+        there may be no duplicate entries between these lists.
+    * If a local {{SanitizerElementNamespaceWithAttributes/removeAttributes}} remove lists exists,
+        there may be no duplicate entries between these lists.
+    * {{SanitizerConfig/dataAttributes}} must be absent.
+
+
+Note that these rules put a somewhat higher burden on configuration authors, as it's on them
+to avoid configurations which might be ambiguous.
+
+<div algorithm>
+A {{SanitizerConfig}} |config| is <dfn for=SanitizerConfig>valid</dfn> if all of the following
+conditions hold:
+
+1. The |config| has an {{SanitizerConfig/elements}} or an {{SanitizerConfig/removeElements}}
+    [=map/key=], or neither, but not both.
+1. The |config| has an {{SanitizerConfig/attributes}} or an {{SanitizerConfig/removeAttributes}}
+    [=map/key=], or neither, but not both.
+1. Assuming that all element names have run through [=canonicalize a sanitizer element=] algorithm,
+    and all attribute names have run through [=canonicalize a sanitizer attribute=] algorithm.
+    1. The [=set/difference=] of |config|[{{SanitizerConfig/elements}}] and
+        |config|[{{SanitizerConfig/removeElements}}] is [=list/empty=].
+    1. The [=set/difference=] of |config|[{{SanitizerConfig/elements}}] and
+        |config|[{{SanitizerConfig/replaceWithChildrenElements}}] is [=list/empty=].
+    1. The [=set/difference=] of |config|[{{SanitizerConfig/removeElements}}] and
+        |config|[{{SanitizerConfig/replaceWithChildrenElements}}] is [=list/empty=].
+    1. The [=set/difference=] of |config|[{{SanitizerConfig/attributes}}] and
+        |config|[{{SanitizerConfig/removeAttributes}}] is [=list/empty=].
+    1. If |config|[{{SanitizerConfig/attributes}}] [=map/exists=]:
+        1. [=list/iterate|For any=] |element| in |config|[{{SanitizerConfig/elements}}]:
+            1. The [=set/difference=] of |config|[{{SanitizerConfig/attributes}}] and
+                |element|[{{SanitizerElementNamespaceWithAttributes/attributes}}] is
+                [=list/empty=].
+            1. The [=set/difference=] of |config|[{{SanitizerConfig/attributes}}] and
+                |element|[{{SanitizerElementNamespaceWithAttributes/removeAttributes}}] is
+                [=list/empty=].
+            1. If {{SanitizerConfig/dataAttributes}} is true:
+                1. Neither |config|[{{SanitizerConfig/attributes}}] nor
```

Comment:
We need to check `|config|[{{SanitizerConfig/attributes}}]` outside the `For any` loop, in case `elements` is empty.